user_tools/src/spark_rapids_tools/cmdli/argprocessor.py

```diff
@@ -663,3 +663,19 @@ def build_tools_args(self) -> dict:
             'n_trials': self.n_trials,
             'platformOpts': {},
         }
+
+
+@dataclass
+@register_tool_arg_validator('instance_description')
+class InstanceDescriptionUserArgModel(AbsToolUserArgModel):
+    """
+    Represents the arguments to run the generate_instance_description tool.
+    """
+
+    def build_tools_args(self) -> dict:
+        runtime_platform = CspEnv.fromstring(self.platform)
```

Comment:
It does not look like the argProcessor is validating the platform input.
For example, this relies on the constructor which by default validates that the ENUM of platform is correct.
However, based on the PR description, not all the platforms should be considered. 

> defines one of the following "dataproc", "emr", and "databricks-azure"

```
$> spark_rapids_dev generate_instance_description --platform dataprocs

2024-06-27 10:05:16,586 ERROR spark_rapids_tools.argparser: Validation err: 1 validation error for InstanceDescriptionUserArgModel
platform
  Input should be 'databricks_aws','databricks_azure','dataproc','dataproc_gke','emr','onprem' or 'NONE' [type=enum, input_value='dataprocs', input_type=str]

ERROR: Could not consume arg: instance_description
Usage: spark_rapids <command>
  available commands:    prediction | profiling | qualification | train

For detailed information on this command, run:
  spark_rapids --help

```